File: solutions/go/isogram/1/isogram.go
package isogram

func IsIsogram(word string) bool {
	var isFound bool = true
	var n []rune
	for _, w := range word {
		if 'A' <= w && w <= 'Z' {
			w += 32
		}
		for _, w1 := range n {
			switch {
			case w == 45 || w == 32:
				isFound = true
			case w == w1:
				isFound = false
			}
		}
		n = append(n, w)
	}
	return isFound
}

File: solutions/c/isogram/1/isogram.c
#include "isogram.h"
#include "string.h" 
#include <ctype.h>

bool is_isogram(const char phrase[]) {
    if (phrase == NULL) {return false;}
    char currentchar = tolower(phrase[0]);
    bool value = true;
    for (size_t i = 1; i < strlen(phrase); i++) {
        for (size_t ii = i; ii < strlen(phrase); ii++) {
            if (tolower(phrase[ii]) == '-') {
                continue;
            } if (tolower(phrase[ii]) == ' ') {
                continue;
            } else if (tolower(phrase[ii]) == currentchar) {
                value = false;
                break;
            }
        }
        currentchar = tolower(phrase[i]);
    }
    return value;
}